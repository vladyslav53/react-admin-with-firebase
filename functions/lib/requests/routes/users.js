"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const express_1 = require("express");
const firebase_admin_1 = require("firebase-admin");
const cors = require('cors')({ origin: true });
const { v4: uuid } = require('uuid');
const router = express_1.Router();
const createUserAuth = async (email, isAdmin) => {
    const { uid } = await firebase_admin_1.auth().createUser({ email, password: uuid() });
    return firebase_admin_1.auth().setCustomUserClaims(uid, {
        isAdmin
    });
};
router.post('/', (request, response) => {
    cors(request, response, async () => {
        const { email, isAdmin } = request.body;
        if (!email) {
            return response
                .status(400)
                .json({ error: { code: 'auth/invalid-email' } });
        }
        let uid;
        try {
            uid = await createUserAuth(email, isAdmin);
        }
        catch (error) {
            return response.status(500).json({ error });
        }
        return response.status(200).json({ uid });
    });
});
exports.default = router;
//# sourceMappingURL=users.js.map