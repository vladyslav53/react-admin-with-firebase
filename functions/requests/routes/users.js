const express = require('express');
const admin = require('firebase-admin');
const uuid = require('uuid/v4');

const router = express.Router();

const createUserAuth = async (email, isAdmin) => {
  const { uid } = await admin.auth().createUser({ email, password: uuid() });

  await admin.auth().setCustomUserClaims(uid, {
    isAdmin
  });

  return uid;
};

router.post('/', async (request, response) => {
  const { email, isAdmin } = request.body;

  if (!email) {
    return response.status(400).json({ error: { code: 'auth/invalid-email' } });
  }

  let uid;
  try {
    uid = await createUserAuth(email, isAdmin);
  } catch (error) {
    console.error('Error while creating user', error);
    return response.status(500).json({ error });
  }

  return response.status(200).json({ uid });
});

module.exports = router;
