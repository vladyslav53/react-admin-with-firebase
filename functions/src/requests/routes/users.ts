import { Router } from 'express';
import { auth } from 'firebase-admin';

const cors = require('cors')({ origin: true });
const { v4: uuid } = require('uuid');

const router = Router();

const createUserAuth = async (email: string, isAdmin: boolean) => {
  const { uid } = await auth().createUser({ email, password: uuid() });

  return auth().setCustomUserClaims(uid, {
    isAdmin
  });
};

router.post('/', (request, response) => {
  cors(request, response, async () => {
    const { email, isAdmin } = request.body;

    if (!email) {
      return response
        .status(400)
        .json({ error: { code: 'auth/invalid-email' } });
    }

    let uid;
    try {
      uid = await createUserAuth(email, isAdmin);
    } catch (error) {
      return response.status(500).json({ error });
    }

    return response.status(200).json({ uid });
  });
});

export default router;
