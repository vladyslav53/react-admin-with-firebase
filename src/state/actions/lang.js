import { createAction } from 'redux-act';

export const LANG_SET_LOCALE = createAction('LANG_SET_LOCALE');

export const langSetLocale = locale => dispatch => {
  return dispatch(LANG_SET_LOCALE({ locale }));
};
