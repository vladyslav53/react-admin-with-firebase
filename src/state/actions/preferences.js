import { createAction } from 'redux-act';

export const PREFERENCES_SET_LOCALE = createAction('PREFERENCES_SET_LOCALE');

export const preferencesSetLocale = locale => dispatch => {
  return dispatch(PREFERENCES_SET_LOCALE({ locale }));
};
