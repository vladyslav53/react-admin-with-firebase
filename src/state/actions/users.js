import { createAction } from 'redux-act';
import uuid from 'uuid/v4';
import { toastr } from 'react-redux-toastr';

import axios from 'utils/axios';
import { firebaseError } from 'utils';
import firebase from 'firebase.js';
import { checkUserData, AUTH_UPDATE_USER_DATA } from './auth';

export const USERS_FETCH_DATA_INIT = createAction('USERS_FETCH_DATA_INIT');
export const USERS_FETCH_DATA_SUCCESS = createAction(
  'USERS_FETCH_DATA_SUCCESS'
);
export const USERS_FETCH_DATA_FAIL = createAction('USERS_FETCH_DATA_FAIL');

export const USERS_DELETE_USER_INIT = createAction('USERS_DELETE_USER_INIT');
export const USERS_DELETE_USER_SUCCESS = createAction(
  'USERS_DELETE_USER_SUCCESS'
);
export const USERS_DELETE_USER_FAIL = createAction('USERS_DELETE_USER_FAIL');

export const USERS_CLEAR_DATA = createAction('USERS_CLEAR_DATA');

export const USERS_CREATE_USER_INIT = createAction('USERS_CREATE_USER_INIT');
export const USERS_CREATE_USER_SUCCESS = createAction(
  'USERS_CREATE_USER_SUCCESS'
);
export const USERS_CREATE_USER_FAIL = createAction('USERS_CREATE_USER_FAIL');

export const USERS_MODIFY_USER_INIT = createAction('USERS_MODIFY_USER_INIT');
export const USERS_MODIFY_USER_SUCCESS = createAction(
  'USERS_MODIFY_USER_SUCCESS'
);
export const USERS_MODIFY_USER_FAIL = createAction('USERS_MODIFY_USER_FAIL');

export const USERS_CLEAN_UP = createAction('USERS_CLEAN_UP');

export const fetchUsers = () => {
  return async (dispatch, getState) => {
    dispatch(checkUserData());

    dispatch(USERS_FETCH_DATA_INIT());

    const { tenant, id } = getState().auth.userData;

    let users;

    try {
      const ref = firebase.database().ref('users');

      users = (
        await ref
          .orderByChild('tenant')
          .equalTo(tenant)
          .once('value')
      ).val();
    } catch (error) {
      toastr.error('', error);
      return dispatch(USERS_FETCH_DATA_FAIL({ error }));
    }

    const usersData = users
      ? Object.entries(users).map(([key, value]) => ({
          id: key,
          ...value
        }))
      : [];

    return dispatch(
      USERS_FETCH_DATA_SUCCESS({
        users: usersData.filter(user => user.id !== id)
      })
    );
  };
};

export const deleteUser = id => {
  return async dispatch => {
    dispatch(USERS_DELETE_USER_INIT());

    const user = firebase.auth().currentUser;

    const userToken = await user.getIdToken();

    try {
      await axios(userToken).delete(`/users/${id}`);
    } catch (error) {
      toastr.error('', error);
      return dispatch(USERS_DELETE_USER_FAIL({ error }));
    }

    toastr.success('', 'The user was deleted.');
    return dispatch(USERS_DELETE_USER_SUCCESS({ id }));
  };
};

export const clearUsersData = () => {
  return dispatch => {
    dispatch(USERS_CLEAR_DATA());
  };
};

export const createUser = ({
  name,
  email,
  location,
  file,
  createdAt,
  isAdmin
}) => {
  return async (dispatch, getState) => {
    dispatch(USERS_CREATE_USER_INIT());

    const { tenant } = getState().auth.userData;

    const user = firebase.auth().currentUser;

    const userToken = await user.getIdToken();

    const body = new FormData();

    if (file) {
      const fileExtension = file.name.split('.')[1];

      const fileName = `${uuid()}.${fileExtension}`;

      body.append('logo', file, fileName);
    }

    body.append('name', name);
    body.append('location', location);
    body.append('email', email);
    body.append('password', uuid());
    body.append('tenant', tenant);
    body.append('createdAt', createdAt);
    body.append('isAdmin', isAdmin);

    axios(userToken)
      .post('/users', body)
      .then(async response => {
        const userCreated = response.data;

        const actionCodeSettings = {
          url: process.env.REACT_APP_LOGIN_PAGE_URL,
          handleCodeInApp: true
        };

        try {
          await firebase
            .auth()
            .sendSignInLinkToEmail(email, actionCodeSettings);
        } catch (error) {
          const errorMessage = firebaseError(error.response.data.error.code);

          return dispatch(
            USERS_CREATE_USER_FAIL({
              error: errorMessage
            })
          );
        }

        toastr.success('', 'User created successfully');
        return dispatch(
          USERS_CREATE_USER_SUCCESS({
            user: userCreated
          })
        );
      })
      .catch(error => {
        const errorMessage = firebaseError(error.response.data.error.code);
        toastr.error('', errorMessage);
        return dispatch(
          USERS_CREATE_USER_FAIL({
            error: errorMessage
          })
        );
      });
  };
};

export const modifyUser = ({
  name,
  location,
  isAdmin,
  file,
  createdAt,
  id,
  isEditing,
  isProfile
}) => {
  return async (dispatch, getState) => {
    dispatch(USERS_MODIFY_USER_INIT());

    const { tenant } = getState().auth.userData;

    const user = firebase.auth().currentUser;

    const userToken = await user.getIdToken();

    const body = new FormData();

    if (file) {
      const fileExtension = file.name.split('.')[1];

      const fileName = `${uuid()}.${fileExtension}`;

      body.append('logo', file, fileName);
    }

    body.append('name', name);
    body.append('location', location);
    body.append('tenant', tenant);
    body.append('createdAt', createdAt);
    body.append('isAdmin', isAdmin);

    axios(userToken)
      .patch(`/users/${id}`, body)
      .then(response => {
        const userCreated = response.data;
        const { uid } = firebase.auth().currentUser;

        if (id === uid) {
          dispatch(AUTH_UPDATE_USER_DATA({ ...userCreated }));
        }

        if (isProfile) {
          toastr.success('', 'Profile updated successfully');
        } else if (isEditing) {
          toastr.success('', 'User updated successfully');
        }

        return dispatch(
          USERS_MODIFY_USER_SUCCESS({
            user: userCreated
          })
        );
      })
      .catch(error => {
        const errorMessage = firebaseError(error.response.data.error.code);
        toastr.error('', errorMessage);
        return dispatch(
          USERS_MODIFY_USER_FAIL({
            error: errorMessage
          })
        );
      });
  };
};

export const usersCleanUp = () => dispatch => dispatch(USERS_CLEAN_UP());
