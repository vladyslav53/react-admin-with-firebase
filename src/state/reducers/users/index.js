import { createReducer } from 'redux-act';

import {
  USERS_FETCH_DATA_INIT,
  USERS_FETCH_DATA_SUCCESS,
  USERS_FETCH_DATA_FAIL,
  USERS_DELETE_USER_INIT,
  USERS_DELETE_USER_SUCCESS,
  USERS_DELETE_USER_FAIL,
  USERS_CLEAR_DATA,
  USERS_CREATE_USER_INIT,
  USERS_CREATE_USER_SUCCESS,
  USERS_CREATE_USER_FAIL,
  USERS_MODIFY_USER_INIT,
  USERS_MODIFY_USER_SUCCESS,
  USERS_MODIFY_USER_FAIL,
  USERS_CLEAN_UP,
  USERS_CLEAR_DATA_LOGOUT,
} from 'state/actions/users';

const initialState = {
  list: [],
  user: {
    name: '',
    email: '',
    location: '',
    isAdmin: false,
    file: null,
    createdAt: new Date().toDateString(),
  },
  loading: false,
  error: null,
  success: false,
  deleted: false,
};

export const usersReducer = createReducer(
  {
    [USERS_FETCH_DATA_INIT]: () => ({
      ...initialState,
      loading: true,
    }),
    [USERS_FETCH_DATA_SUCCESS]: (state, payload) => ({
      ...state,
      list: payload.list || [],
      user: payload.user || initialState.user,
      loading: false,
      error: null,
    }),
    [USERS_FETCH_DATA_FAIL]: (state, payload) => ({
      ...state,
      loading: false,
      error: payload.error,
    }),
    [USERS_DELETE_USER_INIT]: (state) => ({
      ...state,
      loading: true,
    }),
    [USERS_DELETE_USER_SUCCESS]: (state, payload) => ({
      ...state,
      list: state.list.filter((elem) => elem.id !== payload.id),
      loading: false,
      error: null,
      deleted: true,
    }),
    [USERS_DELETE_USER_FAIL]: (state, payload) => ({
      ...state,
      loading: false,
      error: payload.error,
    }),
    [USERS_CLEAR_DATA]: (state) => ({
      ...initialState,
      list: state.list,
    }),
    [USERS_CREATE_USER_INIT]: (state) => ({
      ...state,
      loading: true,
    }),
    [USERS_CREATE_USER_SUCCESS]: (state, payload) => ({
      ...state,
      list: state.list.concat(payload.user),
      loading: false,
      error: null,
      success: true,
    }),
    [USERS_CREATE_USER_FAIL]: (state, payload) => ({
      ...state,
      loading: false,
      error: payload.error,
    }),
    [USERS_MODIFY_USER_INIT]: (state) => ({
      ...state,
      loading: true,
    }),
    [USERS_MODIFY_USER_SUCCESS]: (state, payload) => ({
      ...state,
      list: !state.list
        ? []
        : state.list.map((elem) => {
            if (elem.id === payload.id) {
              return {
                name: payload.user.name,
                location: payload.user.location,
                id: payload.id,
                logoUrl: payload.user.logoUrl,
                createdAt: payload.user.createdAt,
                email: elem.email,
              };
            }
            return elem;
          }),
      loading: false,
      error: null,
      success: true,
    }),
    [USERS_MODIFY_USER_FAIL]: (state, payload) => ({
      ...state,
      loading: false,
      error: payload.error,
    }),
    [USERS_CLEAN_UP]: (state) => ({
      ...state,
      loading: false,
      error: null,
      success: false,
      deleted: false,
    }),
    [USERS_CLEAR_DATA_LOGOUT]: () => ({
      ...initialState,
    }),
  },
  initialState
);
