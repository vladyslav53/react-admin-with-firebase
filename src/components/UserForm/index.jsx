/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import DatePicker from '../DatePicker';
import { useChangeHandler } from '../../utils/hooks';
import { validateEmail } from '../../utils';
import './UserForm.scss';
import { usersCleanUp } from '../../state/actions/users';
import paths from '../../pages/Router/paths';

const UserForm = ({ isEditing, isProfile, userData, action }) => {
  const { loading } = useSelector(
    state => ({
      loading: state.users.loading
    }),
    shallowEqual
  );

  const dispatch = useDispatch();

  useEffect(() => {
    return () => dispatch(usersCleanUp());
  }, [dispatch]);

  const [user, setUser] = useState(userData);

  const onChangeHandler = useChangeHandler(setUser);

  const onDateChangedHandler = createdAt =>
    setUser(prevState => ({
      ...prevState,
      createdAt: createdAt.toDateString()
    }));

  const onFileChangedHandler = event => {
    const file = event.target.files[0];
    setUser(prevState => ({ ...prevState, file, logoUrl: null }));
  };

  const onSubmitHandler = event => {
    event.preventDefault();
    dispatch(
      action({ ...user, createdAt: user.createdAt, isEditing, isProfile })
    );
  };

  let emailInput = {
    modifier: null,
    message: { modifier: null, content: null }
  };

  const invalidEmail = user.email && !validateEmail(user.email);

  if (invalidEmail) {
    emailInput = {
      modifier: 'is-danger',
      message: {
        modifier: 'is-danger',
        content: 'Invalid E-mail'
      }
    };
  }

  const canSubmit =
    user.name && user.location && user.createdAt && !invalidEmail;

  const imagePreviewUrl = !user.logoUrl
    ? user.file && URL.createObjectURL(user.file)
    : user.logoUrl;

  return (
    <>
      <div className="tile is-ancestor">
        <div className="tile is-parent">
          <div className="card tile is-child">
            <header className="card-header">
              <p className="card-header-title">
                <span className="icon">
                  <i className="mdi mdi-account-edit default" />
                </span>
                User Information
              </p>
            </header>
            <div className="card-content">
              <form onSubmit={onSubmitHandler}>

              {isEditing ? (
                <div className="field is-horizontal">
                  <div className="field-label is-normal">
                    <label className="label">E-mail</label>
                  </div>
                  <div className="field-body">
                    <div className="field">
                      <div className="control">
                        <input
                          type="text"
                          readOnly="readOnly"
                          className="input is-static"
                          value={user.email}
                        />
                      </div>
                    </div>
                  </div>
                </div>  
              ) : (
                  <div className="field is-horizontal">
                    <div className="field-label is-normal">
                      <label className="label">E-mail</label>
                    </div>
                    <div className="field-body">
                      <div className="field">
                        <div className="control">
                          <input
                            className={`input ${emailInput.modifier}`}
                            type="email"
                            required
                            name="email"
                            value={user.email}
                            onChange={onChangeHandler}
                          />
                        </div>
                        {emailInput.message.content && (
                          <p
                            className={`help is-${emailInput.message.modifier}`}
                          >
                            {emailInput.message.content}
                          </p>
                        )}
                      </div>
                    </div>
                  </div>
                )}

                <div className="field is-horizontal">
                  <div className="field-label is-normal">
                    <label className="label">Name</label>
                  </div>
                  <div className="field-body">
                    <div className="field">
                      <div className="control">
                        <input
                          id="name"
                          className="input"
                          type="text"
                          required
                          name="name"
                          value={user.name}
                          onChange={onChangeHandler}
                        />
                      </div>
                    </div>
                  </div>
                </div>

                <div className="field is-horizontal">
                  <div className="field-label is-normal">
                    <label className="label">Location</label>
                  </div>
                  <div className="field-body">
                    <div className="field">
                      <div className="control">
                        <input
                          className="input"
                          type="text"
                          name="location"
                          required
                          value={user.location}
                          onChange={onChangeHandler}
                        />
                      </div>
                    </div>
                  </div>
                </div>

                {!isProfile && (
                  <div className="field has-check is-horizontal">
                    <div className="field-label">
                      <label className="label">Admin</label>
                    </div>
                    <div className="field-body">
                      <div className="field">
                        <div className="field">
                          <div className="control">
                            <label className="b-checkbox checkbox">
                              <input
                                type="checkbox"
                                name="isAdmin"
                                onChange={onChangeHandler}
                                checked={user.isAdmin}
                              />
                              <span className="check is-primary" />
                            </label>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                )}

                <div className="field is-horizontal">
                  <div className="field-label is-normal">
                    <label className="label">Created</label>
                  </div>
                  <div className="field-body">
                    <div className="field">
                      <DatePicker
                        date={new Date(user.createdAt)}
                        onChange={onDateChangedHandler}
                      />
                    </div>
                  </div>
                </div>

                <hr />

                <div className="field is-horizontal">
                  <div className="field-label is-normal">
                    <label className="label">Logo</label>
                  </div>
                  <div className="field-body">
                    <div className="field">
                      <div className="file has-name">
                        <label className="file-label">
                          <input
                            className="file-input"
                            type="file"
                            accept="image/*"
                            onChange={onFileChangedHandler}
                          />
                          <span className="file-cta">
                            <span className="file-icon">
                              <i className="fas fa-upload" />
                            </span>
                            <span className="file-label">
                              {user.file ? 'Pick another file' : 'Pick a file'}
                            </span>
                          </span>
                          <span className="file-name">
                            {user.file && user.file.name}
                          </span>
                        </label>
                      </div>
                    </div>
                  </div>
                </div>

                <hr />
                <div className="field is-horizontal">
                  <div className="field-label" />
                  <div className="field-body">
                    <div className="field">
                      <div className="field is-grouped">
                        <div className="control">
                          <button
                            type="submit"
                            className={`button is-primary ${loading &&
                              'is-loading'}`}
                            disabled={!canSubmit}
                          >
                            <span>Submit</span>
                          </button>
                        </div>
                        { !isProfile && (
                          <Link to={paths.USERS} className="button">
                            Go Back
                          </Link>
                        )}
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div className="tile is-parent preview">
          <div className="card tile is-child">
            <header className="card-header">
              <p className="card-header-title">
                <span className="icon">
                  <i className="mdi mdi-account default" />
                </span>
                User Preview
              </p>
            </header>
            <div className="card-content">
              {imagePreviewUrl && (
                <>
                  <div className="is-user-avatar image has-max-width is-aligned-center">
                    <img
                      className="user-avatar"
                      src={imagePreviewUrl}
                      alt="User profile logo preview"
                    />
                  </div>
                  <hr />
                </>
              )}

              {!isEditing && (
                <div className="field">
                  <label className="label">E-mail</label>
                  <div className="control is-clearfix">
                    <input
                      type="text"
                      readOnly="readOnly"
                      className="input is-static"
                      value={user.email}
                    />
                  </div>
                </div>
              )}

              <div className="field">
                <label className="label">Name</label>
                <div className="control is-clearfix">
                  <input
                    type="text"
                    readOnly="readOnly"
                    className="input is-static"
                    value={user.name}
                  />
                </div>
              </div>

              <div className="field">
                <label className="label">Location</label>
                <div className="control is-clearfix">
                  <input
                    type="text"
                    readOnly="readOnly"
                    className="input is-static"
                    value={user.location}
                  />
                </div>
              </div>

              {!isProfile && (
                <div className="field">
                  <label className="label">Admin</label>
                  <div className="control is-clearfix">
                    {user.isAdmin ? (
                      <span className="icon">
                        <i className="mdi mdi-check" />
                      </span>
                    ) : (
                      <span className="icon">
                        <i className="mdi mdi-close" />
                      </span>
                    )}
                  </div>
                </div>
              )}

              <div className="field">
                <label className="label">Created</label>
                <div className="control is-clearfix">
                  <input
                    type="text"
                    readOnly="readOnly"
                    className="input is-static"
                    value={user.createdAt}
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

UserForm.propTypes = {
  isEditing: PropTypes.bool,
  userData: PropTypes.shape({
    id: PropTypes.string,
    isAdmin: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
    logoUrl: PropTypes.string,
    tenant: PropTypes.string,
    createdAt: PropTypes.string.isRequired
  }),
  action: PropTypes.func.isRequired
};

export default UserForm;
