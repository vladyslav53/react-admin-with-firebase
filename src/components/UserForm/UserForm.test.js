import React from 'react';

import * as actions from 'state/actions/users';
import UserForm from '.';

describe('<UserForm /> rendering', () => {
  let userData;
  beforeEach(() => {
    userData = {
      email: 'mkrukuy@gmail.com',
      name: 'Mateo',
      location: 'Montevideo, Uruguay',
      isAdmin: false,
      file: null,
      id: 'test id',
      logoUrl: 'some logoUrl',
      createdAt: new Date().toDateString(),
    };
  });

  it('should render without crashing', () => {
    const user = { ...userData, createdAt: '11/21/2020' };

    const { component } = shallowWithProviders(
      <UserForm user={user} action={actions.createUser} />
    )({
      users: {},
    });

    expect(component).toMatchSnapshot();
  });

  it('should display user name preview', () => {
    const { component } = mountWithProviders(
      <UserForm user={userData} action={actions.createUser} />
    )({
      users: {},
    });

    expect(component.find('input.input.is-static').at(1).props().value).toEqual(
      'Mateo'
    );
  });

  it('should display email preview if it is creating a new user', () => {
    const { component } = mountWithProviders(
      <UserForm user={userData} action={actions.createUser} />
    )({
      users: {},
    });

    expect(
      component.find('input.input.is-static').first().props().value
    ).toEqual('mkrukuy@gmail.com');
  });

  it('should display location preview', () => {
    const { component } = mountWithProviders(
      <UserForm user={userData} isEditing action={actions.createUser} />
    )({
      users: {},
    });

    expect(component.find('input.input.is-static').at(2).props().value).toEqual(
      'Montevideo, Uruguay'
    );
  });

  it('should display admin preview', () => {
    const { component } = mountWithProviders(
      <UserForm user={userData} isEditing action={actions.createUser} />
    )({
      users: {},
    });

    expect(component.exists('.icon')).toBeTruthy();
  });

  it('should display created preview', () => {
    const { component } = mountWithProviders(
      <UserForm user={userData} isEditing action={actions.createUser} />
    )({
      users: {},
    });
    expect(component.find('p.date')).toBeTruthy();
  });
});
