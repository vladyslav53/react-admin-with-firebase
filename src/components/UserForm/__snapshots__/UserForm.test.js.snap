// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm /> rendering should render without crashing 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <UserForm
      action={[Function]}
      userData={
        Object {
          "createdAt": "11/21/2020",
          "email": "mkrukuy@gmail.com",
          "file": null,
          "id": "test id",
          "isAdmin": false,
          "location": "Montevideo, Uruguay",
          "logoUrl": "some logoUrl",
          "name": "Mateo",
          "tenant": "some tenantId",
        }
      }
    />
  </Provider>
</Router>
`;
